### PR TITLE
Adding the ability to diable CSTG key

### PR DESCRIPTION
**Pull request by cody-constine-ttd**

### CODE REVIEW

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 1]
```diff
@@ -3094,6 +3093,55 @@ void cstgLogsInvalidAppName(String appName, Vertx vertx, VertxTestContext testCo
                 });
     }
 
+    @Test
+    void catalogsDisabledAsUnauthorized(Vertx vertx, VertxTestContext testContext) throws NoSuchAlgorithmException, InvalidKeyException {
```
Comment (Ian-Nara) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:3097]: can it be prefixed with "cstg" to make the tests easier to search and group?

File: src/main/java/com/uid2/operator/monitoring/TokenResponseStatsCollector.java [hunk 2]
```diff
@@ -38,7 +38,8 @@ public enum ResponseStatus {
         PayloadHasNoBody,
         /* End of CSTG-related Status */
         Unknown,
-        NoActiveKey
+        NoActiveKey,
+        UNAUTHORIZED
```
Comment (Ian-Nara) [inline src/main/java/com/uid2/operator/monitoring/TokenResponseStatsCollector.java:42]: The others are PascalCase

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 3]
```diff
@@ -3094,6 +3093,55 @@ void cstgLogsInvalidAppName(String appName, Vertx vertx, VertxTestContext testCo
...
+                "v2/token/client-generate",
+                null,
+                requestJson,
+                secretKey,
+                401,
+                testContext,
+                respJson -> {
+                    assertEquals("Unauthorized", respJson.getString("message"));
+                    testContext.completeNow();
+                });
```
Comment (Ian-Nara) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:3142]: I think we can also assertTokenStatusMetrics here

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 4]
```diff
@@ -3094,6 +3093,55 @@ void cstgLogsInvalidAppName(String appName, Vertx vertx, VertxTestContext testCo
...
+        ((Logger) LoggerFactory.getLogger(UIDOperatorVerticle.class)).addAppender(logWatcher);
+        this.uidOperatorVerticle.setLastInvalidOriginProcessTime(Instant.now().minusSeconds(3600));
+
+        setupCstgBackend();
+        String subscriptionID = "PpRrE5YY84";
+        ClientSideKeypair keypairDisabled = new ClientSideKeypair(subscriptionID, clientSideTokenGeneratePublicKey, clientSideTokenGeneratePrivateKey, clientSideTokenGenerateSiteId, "", Instant.now(), true, "");
+        when(clientSideKeypairProvider.getSnapshot()).thenReturn(clientSideKeypairSnapshot);
+        when(clientSideKeypairSnapshot.getKeypair(subscriptionID)).thenReturn(keypairDisabled);
+
+        final KeyFactory kf = KeyFactory.getInstance("EC");
```
Comment (thomasm-ttd) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:3109]: It feels wrong to have to create a fully formed and properly encrypted payload to validate that the key is not authorised. Decrypting the request will be expensive, and we should not be doing that before validating that the keypair is disabled.

**Review by Ian-Nara — APPROVED** [review]

**Review by thomasm-ttd — APPROVED** [review]

**Pull request merged**